### PR TITLE
✨(jest) Take seed coming from Jest whenever provided

### LINKED ISSUE #3287: Prepare `@fast-check/jest` to seeded jest
## 💡 Idea

See https://github.com/facebook/jest/pull/13400 and https://github.com/facebook/jest/pull/12922

Here are some stuffs to be investigated:
- leverage new seed
- do not print the seed anymore, let jest print it (investigate the infinite sync loop case first)
  - do we ask user to set it?
  - do we override it ourselves?

### PR DESCRIPTION
**Pull request by dubzzz**
Fixes #3287

**_Category:_**

- [x] ✨ Introduce new features
- [ ] 📝 Add or update documentation
- [ ] ✅ Add or update tests
- [ ] 🐛 Fix a bug
- [ ] 🏷️ Add or update types
- [ ] ⚡️ Improve performance
- [ ] _Other(s):_ ...
  
  

**_Potential impacts:_**

- [ ] Generated values
- [ ] Shrink values
- [ ] Performance
- [ ] Typings
- [ ] _Other(s):_ ...

### CODE REVIEW

File: packages/jest/src/jest-fast-check.ts [hunk 1]
```diff
@@ -29,7 +29,14 @@ function internalTestPropExecute<Ts extends [any] | any[]>(
     if (seedFromGlobals !== undefined) {
       customParams.seed = seedFromGlobals;
     } else {
-      customParams.seed = Date.now() ^ (Math.random() * 0x100000000);
+      // This option is only available since v29.2.0 of Jest
+      // See official release note: https://github.com/facebook/jest/releases/tag/v29.2.0
+      const seedFromJest = typeof jest.getSeed === 'function' ? jest.getSeed() : undefined;
+      if (seedFromJest) {
```
Comment (dubzzz) [inline packages/jest/src/jest-fast-check.ts:35]:
```suggestion
      if (seedFromJest !== undefined) {
```

**Comment by SimenB** [comment]
Would it make sense to ask people to pass `--show-seed` or add it to their config? Should this module stop printing the seed in its own?

Might be breaking changes, but I'd be very interested to hear your thoughts (as it might mean changes to Jest would still be useful if there are blockers) 🙂 If the answer is "next major" that's understandable as well 👍

**Comment by dubzzz** [comment]
I added a small section recommending to add `--show-seed` into the Readme file. Concerning "not printing the seed via this library", it's definitely part of the plan. But as it would be a breaking I'll wait a bit before doing it.

**Comment by SimenB** [comment]
Cool 👍 Nothing missing in Jest (as of now, regarding this feature)?

**Comment by dubzzz** [comment]
For now everything looks perfect ☺️

**Comment by SimenB** [comment]
🚀

**Comment by dubzzz** [comment]
I just launched a release for `@fast-check/jest`

**Comment by dubzzz** [comment]
Version successfully released, see `@fast-check/jest` version 1.3.0.

**Pull request merged**